### ISSUE TITLE
Plugin Request: Rachio IRO sprinkler system and Ecobee thermostat

### ISSUE THREAD

**Issue by AppleTechy** [nfarina/homebridge#452]
I was hoping someone else out in the Homebridge users has a Rachio Iro and/or an Ecobee Thermostat. I would love if some one would code a plugin in for either of these. I would be more than happy to test them. Both of them have Public documented APIs so it should be fairly easy to build. Thanks for all the help, Apple T

**Comment by pdlove** [comment]
Let me think on this. I currently do Ecobee through Wink and it works great on homebridge-wink. It also natively supports HomeKit, although that disables "Auto" mode.

**Comment by robertvorthman** [comment]
The Ecobee already works great with HomeKit.  It's my most reliable HomeKit device.  It's recent software update has added back the ability to have both "Auto" mode and HomeKit.

**Comment by AppleTechy** [comment]
The issue is I have the revision of the devic from right before they
updated it to have native HomeKit support. So no HomeKit support for the
mode J have. :(

**Comment by Wildcat1** [comment]
@AppleTechy I have a Rachio as well and am also looking to try and get this supported through Homebridge.

**Comment by fitzage** [comment]
Please please please.

I mean, +1

**Comment by AppleTechy** [comment]
Waiting on to get confirmation about WWDC to see if Sprinkler/irrigation
systems are officially supported or if we will just have to implement it as
a switch with basic off and on commands.

**Comment by jpringle03** [comment]
Would love to see this as well 👍  +1

**Comment by apressly** [comment]
Any updates on this one? still waiting for Ratchio Plugin :)

**Comment by spock888** [comment]
"Siri, water backyard for 20 minutes" :point_up_2:

**Comment by Aduffy** [comment]
+1 Please and thank you!

**Comment by vojtamolda** [comment]
I wrote a plugin that exposes the _Ecobee 3_ wireless temperature and occupancy sensors as HomeKit accessories. As was said in the comments above, there's no need to do anything for the thermostat itself since it already is behaves like a HomeKit accessory. However, the sensors are not directly visible in the Home app out of the box.

Here's the repo and a screenshot if anyone is interested: https://github.com/vojtamolda/homebridge-ecobee3-sensors It's not really a critical functionality, but it's fun to have some home automation rules based on occupancy of each room.

![home](https://cloud.githubusercontent.com/assets/8467740/20909277/36532588-bb20-11e6-8be2-5db706656a49.png)

**Comment by kenderson** [comment]
+1 for a Rachio plugin!!

**Comment by mrose17** [comment]
@kenderson & @apressly - rach.io has documented their API -- https://rachio.readme.io/docs/getting-started -- but irrgation systems are not in the latest HomeKit (thanks to @AppleTechy for checking!)

i've started on a rach.io plugin, but i think all i can do for each zone is:

1. use Brightness (0-100) to get/set the time remaining to irrigate in the zone (0 = off, 100 = 30m?) - this makes the zone a lightbulb (or use Volume to make the zone a speaker)

2. use currentPosition/targetPosition (0-100) - this makes the zone a window

3. use ProgrammableSwitchEvent/ProgrammableSwitchOutputState (0/1) to turn watering on or off - this makes the zone a stateful programming switch

anyone have any opinions as to which modeling strategy is the "least bad"???

cc: @nfarina

**Comment by AppleTechy** [comment]
I think the best of the solutions would be to categorize it as a dimmable light switch.. When the light is "on" sprinkler is running.. The dimmable part would. The time it's running, in minutes maybe?? Each percent equals 1 min, 1:1. And then the plugin updates the percentage of the device to what ever the remaining time it left on the cycle, and if you re adjust the percentage on the "light switch" sprinkler. Another solution is just to wait till Irrigarion Controllers are fully supported, hopefully Summer of 2017. As part of the reason I was hoping for HomeKit support would be for the voice controls but the voice controls are all funky when masquerading devices. I appreciate the interest and the work being put into it though!

**Comment by Aduffy** [comment]
@mrose17 Id go with #3... Simple ON/Off, as that is what the Rachio is doing.. Opening and closing a switch. If people want to use HomeKit for scheduling the sprinklers they could do that same as you scheduel your lights coming on and off...  (but not sure why you would do that instead of Rachios internal schedules). I think realistically (for me at least) it would be enough to be able to say  "Turn on/off the front lawn sprinkler" .

**Comment by hadolphs** [comment]
At this point I'm waiting to see what comes out of the upcoming WWDC.  But, has anyone started work on this ? I saw @mrose17 had a repo up at some point but now I can't find it.  Do yo guys know if this just got put on hold?

**Comment by mrose17** [comment]
@hadolphs - there were API issues and i unpublished it (actually, it got published by mistake as well). my goal is to revisit it, when i have some free time, which, sadly, won't be for a while...

**Comment by brownad** [comment]
Use the homeassistant plugin with homebridge which now supports exactly what you want... might be worth the extra setup

**Comment by hadolphs** [comment]
@brownad Thanks for the pointer on home assistant.  I haven't installed it yet on my homebridge, but will definitely try it.  I like the way HA responds NLU-wise.  
Now.. with yesterday's announcements at WWDC '17 ... I noticed that supposedly HomeKit already supports sprinklers ... So it makes me wonder about the earlier conversation in this thread where the working assumption was that HK did not support them... any idea about this? perhaps @mrose17 might know...  Thanks guys

**Comment by brownad** [comment]
Doesn't look like it unless something is listed in the apple shop (not seen a vendor yet)... e.g. Doorbell was shown but august haven't released the video doorbell with HomeKit support yet.

**Comment by hadolphs** [comment]
Yeah it was kind of confusing, because the way he was presenting it, made it seem like sprinklers was already supported (from before) and they were just adding speakers.  I was also stalking this thread in rachio's group and clearly i wasn't the only one who caught that: http://community.rachio.com/t/apple-homekit-for-ios/33/111
Check out these two screen shots, it shows rachio as a vendor (near the lower right corner)... and then sprinklers as a category in homekit:
<img width="1413" alt="homekit partners" src="https://user-images.githubusercontent.com/479196/26841099-74dfa49a-4ab6-11e7-97c8-b3a238a5d54b.png">
<img width="658" alt="homekitcats" src="https://user-images.githubusercontent.com/479196/26841100-74e02bcc-4ab6-11e7-91ec-5188a8edbcb0.png">

**Comment by brownad** [comment]
Future product I would guess

**Comment by AppleTechy** [comment]
Haha interesting. let’s hold off on closing it since it’s a feature request. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
